### PR TITLE
Simple gas optimizations

### PR DESCRIPTION
**Pull request by tbrent**
https://github.com/code-423n4/2023-06-reserve-findings/issues/27

https://github.com/code-423n4/2023-06-reserve-findings/issues/50

### CODE REVIEW

**Comment by akshatmittal** [comment]
I wasn't going to merge these gas changes, but since you've already created a PR I've also included my changes here as well. Most of them were already covered, but I merged my stash just in case something was missed.

File: contracts/p1/BasketHandler.sol [hunk 1]
```diff
@@ -255,7 +255,12 @@ contract BasketHandlerP1 is ComponentP1, IBasketHandler {
 
         for (uint256 i = 0; i < size; ++i) {
             CollateralStatus s = assetRegistry.toColl(basket.erc20s[i]).status();
-            if (s.worseThan(status_)) status_ = s;
+            if (s.worseThan(status_)) {
+                status_ = s;
+            }
+            if (status_ == CollateralStatus.DISABLED) {
```
Comment (tbrent) [inline contracts/p1/BasketHandler.sol:261]: I think this optimization probably increases gas usage on net. There should only ever be a very small number of times this executes with one of the collateral being DISABLED, vs an enormous number of times it executes without DISABLED.

**Comment by tbrent** [comment]
> I wasn't going to merge these gas changes, but since you've already created a PR I've also included my changes here as well. Most of them were already covered, but I merged my stash just in case something was missed.
> 
> Also, seems like the BasetLib changes makes it go "stack too deep"

Sorry for stealing some of the gas optimization work...was looking for something to do and didn't realize it was assigned to you and you were working on it.

I'll take a look at BasketLib...

**Comment by tbrent** [comment]
I ended up reverting the cache'ing in `BasketLib` for the memory arrays but kept some other cache'ing. Also, I'm mad that calling a library function through the "Using For" pattern takes up space on the stack. It shouldn't work that way...

**Comment by akshatmittal** [comment]
Closing in favor of #894

**Pull request closed without merging**